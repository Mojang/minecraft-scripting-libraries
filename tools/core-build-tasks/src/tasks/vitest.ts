import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
