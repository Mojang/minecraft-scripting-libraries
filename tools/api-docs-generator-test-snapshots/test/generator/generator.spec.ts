// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Generator', () => {
    it('Properly generates generator documentation and definitions', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
