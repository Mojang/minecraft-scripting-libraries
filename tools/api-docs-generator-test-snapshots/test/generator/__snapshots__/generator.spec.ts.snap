// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generator > Properly generates generator documentation and definitions > # Of Files Generated 1`] = `4`;

exports[`Generator > Properly generates generator documentation and definitions > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: generator-module
    items:
    - name: generator-module
      href: generator-module/generator-module.md
    - name: ExampleClass
      href: generator-module/ExampleClass.md
"
`;

exports[`Generator > Properly generates generator documentation and definitions > msdocs/scriptapi/generator-module/ExampleClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: generator-module.ExampleClass Class
description: Contents of the generator-module.ExampleClass class.
---
# ExampleClass Class

## Methods
- [methodWhichTakesGenerator](#methodwhichtakesgenerator)

### **methodWhichTakesGenerator**
\`
methodWhichTakesGenerator(generator: Generator<ExampleClass, number, string>): number
\`

#### **Parameters**
- **generator**: Generator<[*ExampleClass*](ExampleClass.md), *number*, *string*>

**Returns** *number*
"
`;

exports[`Generator > Properly generates generator documentation and definitions > msdocs/scriptapi/generator-module/generator-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: generator-module Module
description: Contents of the generator-module module
---
# \`generator-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "generator-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ExampleClass](ExampleClass.md)
"
`;

exports[`Generator > Properly generates generator documentation and definitions > typescript/generator-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "generator-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class ExampleClass {
    private constructor();
    methodWhichTakesGenerator(generator: Generator<ExampleClass, number, string>): number;
}
"
`;
